map/framework.cpp

```diff
@@ -1650,11 +1650,52 @@ void Framework::ConnectToGpsTracker()
 void Framework::DisconnectFromGpsTracker()
 {
   m_connectToGpsTrack = false;
-  GpsTracker::Instance().Disconnect();
+  auto & tracker = GpsTracker::Instance();
+  tracker.Disconnect();
+  tracker.SetEnabled(false);
   if (m_drapeEngine)
     m_drapeEngine->ClearGpsTrackPoints();
 }
 
+void Framework::StartTrackRecording()
+{
+  auto & tracker = GpsTracker::Instance();
+  if (!tracker.IsEnabled())
+    tracker.SetEnabled(true);
+  m_connectToGpsTrack = true;
+  if (m_drapeEngine)
+  {
+    m_drapeEngine->ClearGpsTrackPoints();
+    tracker.Connect(bind(&Framework::OnUpdateGpsTrackPointsCallback, this, _1, _2));
+  }
+}
+
+void Framework::StopTrackRecording()
+{
+  m_connectToGpsTrack = false;
+  auto & tracker = GpsTracker::Instance();
+  tracker.Disconnect();
+  tracker.SetEnabled(false);
+  if (m_drapeEngine)
+    m_drapeEngine->ClearGpsTrackPoints();
+}
+
+void Framework::StopTrackRecordingAndSaveWithName(std::string const & name)
+{
+  GetBookmarkManager().SaveTrackRecording(name);
+  StopTrackRecording();
+}
+
+bool Framework::IsTrackRecordingEmpty() const
```

Comment:
Which UI case requires it?

Reply:
I display a toast that the track is empty and there is nothing to save to prevent calling the `SaveTrackRecording`:

<img width="300" alt="image" src="https://github.com/user-attachments/assets/35f97e65-7739-4493-95f0-78d6201f93c9">

Reply:
nit: Could returning `false` from `StopTrackRecordingAndSaveWithName` be an alternative solution?

Reply:
Returning `false` from the `StopTrackRecordingAndSaveWithName` may be unclear without the stop reason. In this case we should provide the additional documentation describing that stop failing == empty track.